resources/shared/benchmark.mjs

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-case-declarations */
+import { TestRunner } from "./test-runner.mjs";
+import { Params } from "./params.mjs";
+
+/**
+ * BenchmarkStep
+ *
+ * A single test step, with a common interface to interact with.
+ */
+export class BenchmarkStep {
+    constructor(name, run) {
+        this.name = name;
+        this.run = run;
+    }
+
+    async runAndRecord(params, suite, test, callback) {
+        const testRunner = new TestRunner(null, null, params, suite, test, callback);
+        const result = await testRunner.runTest();
```

Comment:
Do we need an explicit await here? Can't we just return the result of `testRunner.runTest()`?

Reply:
yes, since the RAFTestInvoker has this line:

`
const result = await this._reportCallback();
`